draft-ietf-httpbis-message-signatures.md

```diff
@@ -422,37 +481,35 @@ Note that the `inner-list` serialization is used instead of the `sf-list` serial
+3. Sign the signature input to produce the signature output {{sign-sig-input}}
 
 ### Choose and Set Signature Metadata Properties {#choose-metadata}
 
-1. The signer chooses an HTTP Signature Algorithm from those registered in the HTTP Signature Algorithms Registry defined by this document, and sets the signature's Algorithm property to that value.  The signer MUST NOT choose an algorithm marked "Deprecated".  The mechanism by which the signer chooses an algorithm is out of scope for this document.
+1. The signer chooses an HTTP signature algorithm and key material for signing. The signer MUST choose key material that is appropriate 
+    for the signature's algorithm, and that conforms to any requirements defined by the algorithm, such as key size or format. The 
+    mechanism by which the signer chooses the algorithm and key material is out of scope for this document.
 
-2. The signer chooses key material to use for signing and verification, and sets the signature's Verification Key Material property to the key material required for verification.  The signer MUST choose key material that is appropriate for the signature's Algorithm, and that conforms to any requirements defined by the Algorithm, such as key size or format.  The mechanism by which the signer chooses key material is out of scope for this document.
+2. The signer sets the signature's creation time to the current time.
 
-3. The signer sets the signature's Creation Time property to the current time.
-
-4. The signer sets the signature's Expiration Time property to the time at which the signature is to expire, or to undefined if the signature will not expire.
+3. If applicable, the signer sets the signature's expiration time property to the time at which the signature is to expire.
 
 5. The signer creates an ordered list of content identifiers representing the message content and signature metadata to be covered by the signature, and assigns this list as the signature's Covered Content.
-   * Each identifier MUST be one of those defined in Section 2.
-   * This list MUST NOT be empty, as this would result in creating a signature over the empty string.
+   * Each identifier MUST be one of those defined in .
    * Signers SHOULD include `@request-target` in the list.
-   * Signers SHOULD include a date stamp, such as the `date` header. Alternatively, the `created` signature metadata parameter can fulfil this role.
-   * Further guidance on what to include in this list and in what order is out of scope for this document.  However, the list order is significant and once established for a given signature it MUST be preserved for that signature.
-   * Note that the signature metadata is not included in the explicit list of covered content identifiers since its value is always covered.
+   * Signers SHOULD include a date stamp in some form, such as using the `date` header. Alternatively, the `created` signature metadata parameter can fulfil this role.
```

Comment:
I'm not opposed to that, but we'd want to poll the WG and community before making that change in case there's a corner use case that we're missing where that timestamp wouldn't be available for some reason.